backend/controller/controller.go

```diff
@@ -780,21 +782,73 @@ func (s *Service) SendFSMEvent(ctx context.Context, req *connect.Request[ftlv1.S
 		}
 	}
 
-	if destinationVerb == nil {
+	if destinationRef == nil {
 		if len(candidates) > 0 {
 			return nil, connect.NewError(connect.CodeFailedPrecondition,
 				fmt.Errorf("no transition found from state %s for type %s, candidates are %s", instance.CurrentState, eventType, strings.Join(candidates, ", ")))
 		}
 		return nil, connect.NewError(connect.CodeFailedPrecondition, fmt.Errorf("no transition found from state %s for type %s", instance.CurrentState, eventType))
 	}
 
-	err = tx.StartFSMTransition(ctx, instance.FSM, instance.Key, destinationVerb.ToRefKey(), msg.Body)
+	retryCount, minBackoff, maxBackoff, err := s.retryParamsForTransition(fsm, destinationVerb)
+	if err != nil {
+		return nil, connect.NewError(connect.CodeInternal, err)
+	}
+
+	err = tx.StartFSMTransition(ctx, instance.FSM, instance.Key, destinationRef.ToRefKey(), msg.Body, retryCount, minBackoff, maxBackoff)
 	if err != nil {
 		return nil, connect.NewError(connect.CodeInternal, fmt.Errorf("could not start fsm transition: %w", err))
 	}
 	return connect.NewResponse(&ftlv1.SendFSMEventResponse{}), nil
 }
 
+// retryParamsForTransition finds the retry metadata for the given transition and returns the retry count, min and max backoff.
+func (s *Service) retryParamsForTransition(fsm *schema.FSM, verb *schema.Verb) (count int, minBackoff, maxBackoff time.Duration, err error) {
```

Comment:
Maybe we should add this as a method on `schema.FSM`?

Reply:
Tidied that up by also removing the specialized getters for `MinBackoffDuration`/`MaxBackoffDuration` and instead returning a normalized version with `RetryParams` https://github.com/TBD54566975/ftl/pull/1583/commits/0c85daad74839c5207018aafbdd5accf9aa5cc71